aci/resource_aci_vmmctrlrp.go

```diff
@@ -0,0 +1,615 @@
+		relationParam := relationTovmmRsVxlanNs.(string)
+		err = aciClient.CreateRelationvmmRsVxlanNs(vmmCtrlrP.DistinguishedName, relationParam)
+		if err != nil {
+			return err
+		}
+
+	}
+
+	if relationTovmmRsCtrlrPMonPol, ok := d.GetOk("relation_vmm_rs_ctrlr_p_mon_pol"); ok {
+		relationParam := relationTovmmRsCtrlrPMonPol.(string)
+		err = aciClient.CreateRelationvmmRsCtrlrPMonPol(vmmCtrlrP.DistinguishedName, relationParam)
+		if err != nil {
+			return err
+		}
+
+	}
+
+	if relationTovmmRsMgmtEPg, ok := d.GetOk("relation_vmm_rs_mgmt_e_pg"); ok {
+		relationParam := relationTovmmRsMgmtEPg.(string)
+		err = aciClient.CreateRelationvmmRsMgmtEPg(vmmCtrlrP.DistinguishedName, relationParam)
+		if err != nil {
+			return err
+		}
+
+	}
+
+	d.SetId(vmmCtrlrP.DistinguishedName)
+	return resourceAciVMMControllerRead(d, m)
+}
+
+func resourceAciVMMControllerUpdate(d *schema.ResourceData, m interface{}) error {
+	aciClient := m.(*client.Client)
+	desc := d.Get("description").(string)
+
+	name := d.Get("name").(string)
+	VMMDomainDn := d.Get("vmm_domain_dn").(string)
+
+	vmmCtrlrPAttr := models.VMMControllerAttributes{}
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		vmmCtrlrPAttr.Annotation = Annotation.(string)
+	}
+	if DvsVersion, ok := d.GetOk("dvs_version"); ok {
+		vmmCtrlrPAttr.DvsVersion = DvsVersion.(string)
+	}
+	if HostOrIp, ok := d.GetOk("host_or_ip"); ok {
+		vmmCtrlrPAttr.HostOrIp = HostOrIp.(string)
+	}
+	if InventoryTrigSt, ok := d.GetOk("inventory_trig_st"); ok {
+		vmmCtrlrPAttr.InventoryTrigSt = InventoryTrigSt.(string)
+	}
+	if Mode, ok := d.GetOk("mode"); ok {
+		vmmCtrlrPAttr.Mode = Mode.(string)
+	}
+	if MsftConfigErrMsg, ok := d.GetOk("msft_config_err_msg"); ok {
+		vmmCtrlrPAttr.MsftConfigErrMsg = MsftConfigErrMsg.(string)
+	}
+	if MsftConfigIssues, ok := d.GetOk("msft_config_issues"); ok {
+		vmmCtrlrPAttr.MsftConfigIssues = MsftConfigIssues.(string)
+	}
+	if N1kvStatsMode, ok := d.GetOk("n1kv_stats_mode"); ok {
+		vmmCtrlrPAttr.N1kvStatsMode = N1kvStatsMode.(string)
+	}
+	if NameAlias, ok := d.GetOk("name_alias"); ok {
+		vmmCtrlrPAttr.NameAlias = NameAlias.(string)
+	}
+	if Port, ok := d.GetOk("port"); ok {
+		vmmCtrlrPAttr.Port = Port.(string)
+	}
+	if RootContName, ok := d.GetOk("root_cont_name"); ok {
+		vmmCtrlrPAttr.RootContName = RootContName.(string)
+	}
+	if Scope, ok := d.GetOk("scope"); ok {
+		vmmCtrlrPAttr.Scope = Scope.(string)
+	}
+	if SeqNum, ok := d.GetOk("seq_num"); ok {
+		vmmCtrlrPAttr.SeqNum = SeqNum.(string)
+	}
+	if StatsMode, ok := d.GetOk("stats_mode"); ok {
+		vmmCtrlrPAttr.StatsMode = StatsMode.(string)
+	}
+	if VxlanDeplPref, ok := d.GetOk("vxlan_depl_pref"); ok {
+		vmmCtrlrPAttr.VxlanDeplPref = VxlanDeplPref.(string)
+	}
+	vmmCtrlrP := models.NewVMMController(fmt.Sprintf("ctrlr-%s", name), VMMDomainDn, desc, vmmCtrlrPAttr)
+
+	vmmCtrlrP.Status = "modified"
+
+	err := aciClient.Save(vmmCtrlrP)
+
+	if err != nil {
+		return err
+	}
```

Comment:
There should one pre check for all relations. That will check whether relation DN actually exist or not.